### DISCUSSION TITLE
How to dismiss modal on form post and avoid displaying the redirected page on modal?

### DISCUSSION THREAD

**Discussion by ekimkael** [unpoly/unpoly#582] [Q&A]
I'm working with Adonisjs & Unpoly

I open modal with `up-layer="new"` and displayed a form to create categories but when I submit the form the redirected page (list of all categories) is displayed on the modal.

Here is the link attributes:
`<a href="{{ route('admin.categories.create') }}" class="btn btn-primary d-none d-sm-inline-block" up-layer="new" up-class="modal-blur" up-accept-location="{{ route('admin.categories.show') }}" up-on-accepted="up.reload('.page-body')">`

How to avoid it?

**Comment by adam12** [comment]
I think we spoke via email about this. Were you able to enable logging? It can help to see what Unpoly decided to do.

**Comment by ekimkael** [comment]
Yeah, that's me. I have enabled the logging.
here is what i have:

<img width="1440" alt="Screenshot 2024-01-21 at 03 34 38" src="https://github.com/unpoly/unpoly/assets/10372405/92a48fcc-a05d-4132-a01f-2cedef9ecb71">

**Reply by adam12** [reply]
Is `/dashboard/categories` the path that is output in this snippet? `up-accept-location="{{ route('admin.categories.show') }}"`

Can you share the markup that's generated?

**Reply by ekimkael** [reply]
Yeah it's the the path

**Reply by ekimkael** [reply]
I'm on `/dashboard/categories` when the modal open up and I want to go back there when I create a new category

**Reply by adam12** [reply]
I don't see anything wrong with the code example you pasted. Can you confirm what version of Unpoly you're using?

Unpoly didn't accept the subinteraction (it will show a message in the console like `Accepted modal overlay with value`) so perhaps there is something wrong with the page URL that the form submission returns.

Can you show the response from the Network tab for the request to your server and the response from your server?

**Reply by ekimkael** [reply]
I use the 3.7.2 version: https://cdn.jsdelivr.net/npm/unpoly@3.7.2/unpoly.min.js

<img width="1440" alt="Screenshot 2024-01-21 at 04 12 19" src="https://github.com/unpoly/unpoly/assets/10372405/b2132c1e-a801-42cf-b06f-ba9633c5be1c">

<img width="1440" alt="Screenshot 2024-01-21 at 04 13 32" src="https://github.com/unpoly/unpoly/assets/10372405/fd16cc00-1244-44ea-88ac-23e37120b75e">

<img width="1440" alt="Screenshot 2024-01-21 at 04 17 43" src="https://github.com/unpoly/unpoly/assets/10372405/0e6a16d2-7945-49ed-888c-1cf9e018ed00">

**Reply by adam12** [reply]
Do you have an [`[up-main]`](https://unpoly.com/up-main) element inside the markup that's _not_ in the modal? I'm not sure if that could be the cause, but it's the only thing that's standing out to me.

**Reply by ekimkael** [reply]
Here is the markup:
```
<a href="/dashboard/categories/create" class="btn btn-primary d-none d-sm-inline-block" up-layer="new" up-class="modal-blur" up-accept-location="/dashboard/categories/$slug" up-on-accepted="up.reload('.page-body', {focus:':main'})">
  <!-- Download SVG icon from http://tabler-icons.io/i/plus -->
  Add a category
</a>
```

The only place where I have `up-main` is on the dashboard layout

**Reply by adam12** [reply]
That `up-accept-location` looks wrong. Wouldn't you want it to be `up-accept-location="/dashboard/categories"`?

**Reply by ekimkael** [reply]
Yeah but even when it's `up-accept-location="/dashboard/categories"` it doesn't behave as expected.
I just changed it back to `"/dashboard/categories"` and I still have the problem

**Reply by adam12** [reply]
The only other thing I can suggest is forcing the URL to change. I don't see Unpoly "advancing" history, so perhaps there is an issue there.

Instead of redirecting to `/dashboard/categories`, could you try temporarily redirecting to a different URL, and updating your `up-accept-location` accordingly? You'll need to add that temporary endpoint for the new URL but it could return the `/dashboard/categories` content.

Ie. `/dashboard/categories/created`.

If that doesn't work, maybe Henning or someone else can spot the issue.

**Reply by ekimkael** [reply]
Something really strange happened.
I add the same code on other pages and it works pretty well.

I will try what you suggest on categories page and see if it works.
Thank you very very much for your help and super

**Reply by triskweline** [reply]
> I don't see Unpoly "advancing" history, so perhaps there is an issue there.

By default Unpoly will only advance history for GET requests, since only those are reloadable. The screenshot only shows a POST request.

Some ideas you can try:

- Redirect to a GET route after a successful POST.
- Force a history with an `[up-location]` attribute or `X-Up-Location` attribute.
- Use one of the [many other methods to close an overlay](https://unpoly.com/closing-overlays). E.g. use an event.

**Reply by ekimkael** [reply]
Really sorry to answer this late.
There is a redirection(302) when a category is created.

I tried everything you just mentioned but nothing worked

**Reply by triskweline** [reply]
> There is a redirection(302) when a category is created.

The log in the screenshot says it's rendering content from `/dashboard/categories/create`.
